pages/common/tgpt.md

```diff
@@ -0,0 +1,37 @@
+# tgpt
+
+> A terminal-based AI chatbot without the need for API keys.
+> Available providers: `openai`, `opengpts`, `koboldai`, `phind`, `llama2`, `blackboxai`.
+> More information: <https://github.com/aandrew-me/tgpt>.
+
+- Prompt with default provider (GPT-3.5-turbo):
+
+`tgpt "{{prompt_string}}"`
+
+- Start [m]ulti-line interactive mode:
+
+`tgpt --multiline`
+
+- Generate images and save to current directory:
+
+`tgpt --image "{{prompt_string}}"`
+
+- Generate [c]ode using the default provider:
+
+`tgpt --code "{{prompt_string}}"`
+
+- Prompt a specific provider without animations [q]uitely:
+
+`tgpt --provider {{phind}} --quiet --whole "{{prompt_string}}"`
+
+- Generate and execute [s]hell commands using a specific provider (will be prompted for confirmation):
+
+`tgpt --provider {{phind}} --shell "{{prompt_string}}"`
+
+- Prompt with an API key, model, max-length, temperature, and  (required when using `openai` provider):
+
+`tgpt --provider openai --key "{{api_key}}" --model "{{gpt-3.5-turbo}}" --max-length {{10}} --temperature {{0.7}} --top_p {{0.9}} "{{prompt_string}}"`
+
+- Feed a file as additional input pre-prompt:
+
+`cat {{path/to/file.ext}} | tgpt --provider {{blackboxai}} "{{prompt_string}}"`
```

Comment:
```suggestion
`tgpt --provider {{blackboxai}} "{{prompt}}" < {{path/to/file}}`
```
This is an [Useless Use of Cat (UUOC)](https://porkmail.org/era/unix/award#cat), which leaves to some slowdowns like an unnecessary process fork, the file being read by cat and by tgpt too. This gets worse the larger the file is.

Alternatively, you can change it to something like this:
```suggestion
`{{command } | }tgpt --provider {{blackboxai}} "{{prompt_string}}"
```

Or even:
```suggestion
`{{curl link}} | tgpt --provider {{blackboxai}} "{{prompt_string}}"`
```
This would require the description to change to something like "Use `stdin` as an additional input pre-prompt" or "Pipe a command as an additional input pre-prompt"

Also, we use ".ext" when specific(s) file extension(s) are expected. See [style guide#extensions](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#extensions):
> ### Extensions
> 
> - If a particular extension is expected for the file, append it.
>   For example, `unrar x {{path/to/compressed.rar}}`.
> - In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
>   For instance, in `find.md`'s example "Find files by extension" (`find {{path/to/root}} -name '{{*.ext}}'`)
>   using `{{*.ext}}` explains the command without being unnecessarily specific;
>   while in `wc -l {{path/to/file}}` using `{{path/to/file}}` (without extension) is sufficient.

Reply:
this is niceee. thank you for this!

also imo i think the "feed a file with input redirection" would be better to show since it is a specific and clear usecase rather than a general one. I also think that chatbots are more likely to used in that way (or atleast popular to be used in that way), showing `tgpt` is a good alternative for those paid AI chatbots that does the same thing.

will commit in a little while, thank you again!